recipes/geotrans/all/CMakeLists.txt

```diff
@@ -0,0 +1,60 @@
+cmake_minimum_required(VERSION 3.4)
+project(geotrans VERSION 3.8 LANGUAGES CXX)
+####
+# Conan
+####
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+####
+# Directories
+####
+
+set(DTCCDIR "${CMAKE_CURRENT_SOURCE_DIR}/CCS/src/dtcc/CoordinateSystems")
+set(CCSERVICEDIR "${CMAKE_CURRENT_SOURCE_DIR}/CCS/src")
+set(SRCDIR "${CMAKE_CURRENT_SOURCE_DIR}/GEOTRANS3/java_gui/geotrans3/jni")
+
+####
+# Sources
+####
+file(GLOB_RECURSE DTCCSRCS "${DTCCDIR}/*.cpp")
+set(CCSSRCS "${CCSERVICEDIR}/CoordinateConversion/CoordinateConversionService.cpp")
+file(GLOB_RECURSE CCSERVICESRCS "${CCSERVICEDIR}/dtcc/*.cpp")
+list(FILTER CCSERVICESRCS EXCLUDE REGEX ".*${DTCCDIR}.*")
+set(SRCS "${SRCDIR}/strtoval.cpp" "${SRCDIR}/fiomeths.cpp")
+
+####
+# Includes
+####
+
+# Construct DTCC includes
+file(GLOB_RECURSE DTCC_HEADERS "${DTCCDIR}/*.h")
+set(DTTCINCS "")
+foreach(file ${DTCC_HEADERS})
+    get_filename_component(DIR_PATH ${file} PATH)
+    set(DTTCINCS ${DTTCINCS} ${DIR_PATH})
+endforeach()
+list(REMOVE_DUPLICATES DTTCINCS)
+
+# Construct CCS includes. This is all headers not in DTTCINCS
+file(GLOB_RECURSE CCS_HEADERS "${CCSERVICEDIR}/*.h")
+set(INCS ${SRCDIR})
+foreach(file ${CCS_HEADERS})
+    get_filename_component(DIR_PATH ${file} PATH)
+    set(INCS ${INCS} ${DIR_PATH})
+endforeach()
+list(REMOVE_DUPLICATES INCS)
+list(FILTER INCS EXCLUDE REGEX ".*${DTCCDIR}.*")
+
+
+# Create libMSPdtcc
+add_library(MSPdtcc ${DTCCSRCS} ${CCSERVICESRCS})
+target_compile_features(MSPdtcc PUBLIC cxx_std_11)
+target_compile_options(MSPdtcc PRIVATE -pthread -Wno-deprecated)
```

Comment:
Is pthread really needed? C++ threads aren't enough? 
If they are needed, what about the windows case? (there's pthreads for win in CCI)

Reply:
pthread must be linked when using C++11 threads. But there is a more elegant and robust way to link pthread:

```cmake
find_package(Threads REQUIRED)
target_link_libraries(MSPdtcc PRIVATE Threads::Threads)
```